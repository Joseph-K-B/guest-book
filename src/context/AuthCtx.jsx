import React, { createContext, useState } from "react"

const authCtx = createContext();

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  // This would probably be the sign up function 
  // b/c it includes both username and email
  // 
  const login = (username, email, password) => {
    const authConfirmed = (
      username === process.env.REACT_APP_AUTH_USER &&
      email === process.env.REACT_APP_AUTH_EMAIL &&
      password === process.env.REACT_APP_AUTH_PASSWORD
      )
      if(authConfirmed) 
        setUser({ username, email });
      return authConfirmed;
  };

  const logout = (cb) => {
    setUser(null);
    // cb();
  }

  return (
    <authCtx.Provider value={{user, setUser, login, logout}}>
      {children}
    </authCtx.Provider>
  );
};

export {authCtx, AuthProvider};