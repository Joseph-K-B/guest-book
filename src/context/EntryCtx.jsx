import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({children}) => {
    const [note, setNote] = useState('');

    return <NoteCtx.Provider value={{ note, setNote }}>{children}</NoteCtx.Provider>
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}