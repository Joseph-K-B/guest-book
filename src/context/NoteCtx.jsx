import React from 'react';
import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({ children }) => {
    const [notes, setNotes] = useState([]);

    return (
        <NoteCtx.Provider value={{ notes, setNotes }}>{children}</NoteCtx.Provider>
    )
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}