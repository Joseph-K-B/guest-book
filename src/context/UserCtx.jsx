import React from 'react';
import { createContext, useContext, useState } from "react"

const UserCtx = createContext();

const UserProvider = ({children}) => {
    const [name, setName] = useState('');

    return <UserCtx.Provider value={{ name, setName }}>{children}</UserCtx.Provider>
}

const useUser = () => {
    const ctx = useContext(UserCtx);

    if(ctx === undefined) {
        throw new Error('useUser custom hook must be called within child components of UserCtx Provider');
    }
    return ctx;
}

export { UserProvider, useUser }