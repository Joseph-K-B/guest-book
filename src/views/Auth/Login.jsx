import { useState } from "react";
import { useLocation, useNavigate } from "react-router-dom";
import useAuth from "../../hooks/useAuth";
import useForm from "../../hooks/useForm";
import css from './Login.css';


function Login() {
  const {formState, handleFormChange} = useForm({ username: '', email: '', password: ''});
  const [error, setError] = useState(null);
  const navigate = useNavigate();
  const location = useLocation();
  const auth = useAuth();

  const { from } = location.state || {from : {pathname: '/'}};

  const handleLogin = (e) => {
    e.preventDefault();
    const confirmedAuth = auth.login(formState.username, formState.email, formState.password);

    if(confirmedAuth) {
      navigate('/guestbook');
    } else {
      setError('Invalid credentials, form is case sensitive')
    }
  };

  const handleHelp = () => {
    navigate('/help');
  }

  return (
    <>
      <section className={css.formSection}>
        <h3>Register here</h3>
        <form onSubmit={handleLogin} className={css.logForm}>
          <label htmlFor='username'>Name:</label>
          <input
            id='username'
            name='username'
            type='username'
            value={formState.username}
            onChange={(value) => handleFormChange(value)}
          />
          <label htmlFor='email'>Email:</label>
          <input
            id='email'
            name='email'
            type='email'
            value={formState.email}
            onChange={(value) => handleFormChange(value)}
          />
          <label htmlFor='password'>Password:</label>
          <input
            id='password'
            name='password'
            type='password'
            value={formState.password}
            onChange={(value) => handleFormChange(value)}
          />
          <button type='submit' aria-label='sign in' className={css.submitBtn}>
            Sign in
          </button>
        </form>
        {error &&
        <>
          <h4 className={css.error}>{error}</h4>
          <button onClick={() => handleHelp()}>Need help?</button>
        </>
        }
      </section>
    </>
  );
}

export default Login;