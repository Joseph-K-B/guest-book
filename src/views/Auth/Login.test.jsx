import React from 'react';
import Login from "./Login"
import { screen, render } from '@testing-library/react';
import { AuthProvider } from '../../context/AuthCtx';
import { BrowserRouter as Router } from 'react-router-dom';

it('renders log in form and does other cool shit too', () => {
  const {container} = render (
    <AuthProvider>
      <Router>
        <Login />
      </Router>
    </AuthProvider>
  );

  expect(5 + 3).toEqual(8);
  expect(container).toMatchSnapshot();
})