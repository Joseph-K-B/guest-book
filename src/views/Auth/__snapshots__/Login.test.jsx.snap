// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders log in form and does other cool shit too 1`] = `
<div>
  <section
    class="formSection"
  >
    <h3>
      Register here
    </h3>
    <form
      class="logForm"
    >
      <label
        for="username"
      >
        Name:
      </label>
      <input
        id="username"
        name="username"
        type="username"
        value=""
      />
      <label
        for="email"
      >
        Email:
      </label>
      <input
        id="email"
        name="email"
        type="email"
        value=""
      />
      <label
        for="password"
      >
        Password:
      </label>
      <input
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        aria-label="sign in"
        class="submitBtn"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </section>
</div>
`;
